pkg/providers/v1/providers.go

```diff
@@ -186,6 +186,13 @@ type OCI interface {
 	GetAuthenticator() (authn.Authenticator, error)
 }
 
+type Properties interface {
+	Provider
+
+	FetchAllProperties(ctx context.Context, name string, entType minderv1.Entity) (map[string]any, error)
```

Comment:
I'm not sure if "name" is a good name for the parameter and even if string will always be what the provider queries about. We can probably always pass a stringified ID, but IIRC some providers use a numerical ID in their upstream API anyway.